extract.py

```diff
@@ -0,0 +1,139 @@
+"""
+Extracts information from scraped websites, 
+to reduce noise in AI model training.
+"""
+import logging
+import copy
+import regex as re
+from bs4 import BeautifulSoup
+
+class NoSoup(Exception):
+    """
+    Data extractor hasn't generated a soup, and cannot start extracting data.
```

Comment:
Look over this message, why would the extractor want a soup haha?
Make it even more specific by saying that some BeautifulSoup stuff is missing and not soup.